src/pudl/dagster_etl.py

```diff
@@ -0,0 +1,81 @@
+"""Dagster version of EPA CEMS ETL."""
+from pathlib import Path
+
+import pandas as pd
+import sqlalchemy as sa
+from dagster import (AssetMaterialization, EventMetadata, Field, In, Output,
+                     job, op, resource)
+
+import pudl
+from pudl.helpers import compute_dataframe_summary_statistics
+from pudl.metadata.classes import Resource
+from pudl.workspace.datastore import Datastore
+
+# TODO: how can we associate metadata schemas with specific ops?
+EpaCemsDataFrame = Resource.from_id("hourly_emissions_epacems").to_dagster(
+    event_metadata_fn=compute_dataframe_summary_statistics)
+
+
+@op(required_resource_keys={"pudl_settings"}, ins={"transformed_df": In(dagster_type=EpaCemsDataFrame)})
+def load_epacems(context, transformed_df: pd.DataFrame):
+    """Load epacems to parquet."""
+    root_path = Path(context.resources.pudl_settings["parquet_dir"]) / "epacems"
+
+    pudl.load.df_to_parquet(
+        transformed_df,
+        resource_id="hourly_emissions_epacems",
+        root_path=root_path,
+        partition_cols=["year", "state"]
+    )
+
+    # Format of the mapping key makes it difficult to recreate the final parquet path.
+    yield AssetMaterialization(
+        asset_key="hourly_emissions_epacems",
+        description="Persisted result to storage",
+        partition=context.get_mapping_key(),
+        metadata={
+            "Description": f"Parquet file of {context.get_mapping_key()} EPA CEMS data.",
+            "Path": EventMetadata.path(str(root_path)),
+            "Number of Rows": len(transformed_df)
+        },
+    )
+    yield Output(root_path)
```

Comment:
I think the output would be Resource so we could configure it in different envs. 

I'm not sure what would make the most sense to yield here. We could yield the full datastore, a path to the dataset or nothing.

Reply:
Should the load steps be [`@IOManager`](https://docs.dagster.io/concepts/io-management/io-managers)s instead of `@Op`s?

Reply:
Maybe the [`IOManager`](https://docs.dagster.io/concepts/io-management/io-managers) class is useful here?
>The IO manager APIs make it easy to separate code that's responsible for logical data transformation from code that's responsible for reading and writing the results. Ops can focus on business logic, while IO managers handle I/O. This separation makes it easier to test the business logic and run it in different environments.

As for the last `yield`, in [this example](https://docs.dagster.io/tutorial/advanced-tutorial/materializations) they `yield Output(None)` because:
>When we explicitly yield other types of events from op logic [an AssetMaterialization], we need to also explicitly yield [an Output type] so that the framework can recognize them.

Reply:
Yeah I think IOManagers are the mature way to persist data.